src/main/scala/org/zalando/spearheads/innkeeper/api/JsonProtocols.scala

```diff
@@ -62,7 +63,22 @@ object JsonProtocols extends DefaultJsonProtocol {
     }
   }
 
-  implicit val endpointFormat = jsonFormat4(Endpoint.apply)
+  val endpointFormat = jsonFormat(Endpoint.apply, "hostname", "port", "protocol", "type")
+
+  implicit object EndpointFormat extends JsonFormat[Endpoint] {
+    override def write(obj: Endpoint): JsValue = endpointFormat.write(obj)
+
+    override def read(json: JsValue): Endpoint = {
+      val endpoint = endpointFormat.read(json)
+
+      Endpoint(
+        hostname = endpoint.hostname,
+        port = endpoint.port.orElse(Some(443)),
+        protocol = endpoint.protocol.orElse(Some(Https)),
+        endpointType = endpoint.endpointType.orElse(Some(ReverseProxy))
```

Comment:
the problem is that spray-json doesn't offer a way to use default values... so this is kind of a hack... the defaults from the the Endpoint class will only be used for testing... we will never try to create an endpoint in production programmatically...